Source/Events.Store/Actors/EventStore.cs

```diff
@@ -28,220 +25,128 @@ namespace Dolittle.Runtime.Events.Store.Actors;
-            _aggregateRootVersionCache[aggregate] = currentAggregateRootVersion;
-            return CommitForAggregate(request, aggregate, currentAggregateRootVersion, respond);
-        });
-        return Task.CompletedTask;
-
         Task RespondWithFailure(Failure failure)
         {
-            respond(new CommitAggregateEventsResponse { Failure = failure });
+            respond(new CommitEventsResponse { Failure = failure });
             return Task.CompletedTask;
         }
     }
 
-    Task CommitForAggregate(CommitAggregateEventsRequest request, Aggregate aggregate, AggregateRootVersion currentAggregateRootVersion,
-        Action<CommitAggregateEventsResponse> respond)
+    /// <inheritdoc />
+    public override Task CommitForAggregate(CommitAggregateEventsRequest request, Action<CommitAggregateEventsResponse> respond, Action<string> onError)
     {
-        var expectedAggregateRootVersion = request.Events.ExpectedAggregateRootVersion;
-        if (currentAggregateRootVersion != expectedAggregateRootVersion)
+        return ForwardToCommitter<CommitAggregateEventsRequest, CommitAggregateEventsResponse>(request, Respond, RespondWithFailure);
+        Task Respond(CommitAggregateEventsResponse response)
         {
-            return RespondWithFailureAndClearInFlight(new AggregateRootConcurrencyConflict(
-                request.Events.EventSourceId,
-                request.Events.AggregateRootId.ToGuid(),
-                currentAggregateRootVersion,
-                expectedAggregateRootVersion).ToFailure());
-        }
-
-        var tryAdd = Pipeline.TryAddEventsFrom(request);
-        if (!tryAdd.Success)
-        {
-            return RespondWithFailureAndClearInFlight(tryAdd.Exception.ToFailure());
-        }
-
-        var (committedEvents, onNextBatchCompleted) = tryAdd.Result;
-        TrySendBatch();
-        Context.ReenterAfter(onNextBatchCompleted, task =>
-        {
-            if (TryGetFailure(task, out var failure))
-            {
-                return RespondWithFailureAndClearInFlight(failure);
-            }
-
-            _aggregateCommitInFlight.Remove(aggregate);
-            _aggregateRootVersionCache[aggregate] = committedEvents[^1].AggregateRootVersion + 1;
-            respond(new CommitAggregateEventsResponse
-            {
-                Events = committedEvents.ToProtobuf()
-            });
+            respond(response);
             return Task.CompletedTask;
-        });
-        return Task.CompletedTask;
-
-        Task RespondWithFailureAndClearInFlight(Failure failure)
+        }
+        Task RespondWithFailure(Failure failure)
         {
-            _aggregateCommitInFlight.Remove(aggregate);
             respond(new CommitAggregateEventsResponse { Failure = failure });
             return Task.CompletedTask;
         }
     }
 
-    Task<AggregateRootVersion> GetAggregateRootVersion(Aggregate aggregate)
+    Task ForwardToCommitter<TRequest, TResponse>(
```

Comment:
Here you could do a real forward instead of proxying the request.
That would skip sending the response via the eventstore at all

Reply:
Ahhh I see. Like this? [6b915ec](https://github.com/dolittle/Runtime/pull/697/commits/6b915eca28982903cd2e7e60ba941a61b53e1be3)